### PR TITLE
Replace `print()` statements with logging

### LINKED ISSUE #24: CHILLED code is using `print()` statements instead of logging
As mentioned in @glatterf42 's comment [here](https://github.com/iiasa/message-ix-buildings/pull/18#discussion_r1714960117) in #18, it would be better to migrate from using `print()` statements to python's logging.

### PR DESCRIPTION
**Pull request by measrainsey**
Simple PR to replace `print()` statements with python's `logging` functions. 

This should fix #24.

I've implemented a custom logger that:
- creates logs at each module level (`__name__`)
- sets the logging level at `logging.info()`
- saves the logs to a file that is the name of the module (I've added .log files to `.gitignore`)
- adds some additional syntax such as date and time

For example, running `run_preprocess.py` results in the following log files:
- __main__.log
- message_ix_buildings.chilled.preprocess.archetypes.log
- message_ix_buildings.chilled.util.util.log

Hopefully a simple review, but please let me know if I should fix anything.

### CODE REVIEW

File: message_ix_buildings/chilled/util/util.py [hunk 1]
```diff
@@ -22,26 +40,27 @@ def get_archs(config: "Config"):
 
             return archs
         else:
-            print(
+            raise TypeError(
```
Comment (glatterf42) [inline message_ix_buildings/chilled/util/util.py:43]:
```suggestion
            raise FileNotFoundError(
```
Based on the error message, this and all below should probably be `FileNotFoundError`s, shouldn't they?

File: message_ix_buildings/chilled/util/util.py [hunk 2]
```diff
@@ -22,26 +40,27 @@ def get_archs(config: "Config"):
...
+
     elif config.arch_setting == "regional":
         input_file = os.path.join(
             version_path,
             "arch_input_reg.xlsx",
         )
 
         if os.path.exists(input_file):
-            archs = pd.read_excel(input_file, sheet_name="arch").arch.unique()
+            archs = pd.read_excel(input_file, sheet_name="arch").arch.unique()  # type: ignore
```
Comment (glatterf42) [inline message_ix_buildings/chilled/util/util.py:56]: Why do we need `# tpe: ignore` here?

File: message_ix_buildings/chilled/util/util.py [hunk 3]
```diff
@@ -9,6 +10,23 @@ def get_project_root() -> Path:
     return Path(__file__).parent.parent.parent
 
 
+def get_logger(__name__):
+    log = logging.getLogger(__name__)
+    log.setLevel(logging.INFO)
+
+    # configure the handler and formatter as needed
+    handler = logging.FileHandler(f"{__name__}.log", mode="w")
```
Comment (glatterf42) [inline message_ix_buildings/chilled/util/util.py:18]:
```suggestion
def get_logger(name: str):
    log = logging.getLogger(name)
    log.setLevel(logging.INFO)

    # configure the handler and formatter as needed
    handler = logging.FileHandler(f"{name}.log", mode="w")
```
Probably best not to use dunder variables as your own function argument names. These variables have some magic about them that we probably don't want to disturb. 
Fun fact: if you want to indicate that a parameter should only be used internally, call it `_parameter`.

**Review by glatterf42 — APPROVED** [review]
Looks good overall and always a great idea to use python logging over `print()`. Please see my suggestions before merging, though :)

File: message_ix_buildings/chilled/util/util.py (hunk 2 above)
Comment (measrainsey) [reply message_ix_buildings/chilled/util/util.py:56]: We don't -- I shall remove, thank you!

Comment (measrainsey) [reply message_ix_buildings/chilled/util/util.py]: Ah, without the `type: ignore` I was receiving a Mypy error: `

```
Incompatible types in assignment (expression has type "ndarray[Any, Any]", variable has type "list[int | str]").
``` 

I've removed the `type: ignore` and instead added a `.tolist()` to the end of the line, and that seems to have removed the linting error.

**Pull request merged**